From 9e8055faa1a9541b155d9934dc0b03ddb926425f Mon Sep 17 00:00:00 2001
From: aspeedyh <yh_chung@aspeedtech.com>
Date: Tue, 11 Nov 2025 15:13:13 +0800
Subject: [PATCH] eDAF: enable edaf_bridge node in uboot stage

Previous implementation puts edaf_bridge initialization in SPL stage,
use this workaround commit to default enable edaf_bridge node in uboot
stage.

Should be removed after porting edaf_bridge driver to zephyr-rt.

Signed-off-by: aspeedyh <yh_chung@aspeedtech.com>
Change-Id: I4dd2abddd186e44fc6f0d443494cf3ae8710e4de
(cherry picked from commit 84bd8048ad5f6bacec76db96e4e2caba12bf4285)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 arch/arm/dts/ast2700-evb.dts  | 33 +++++++++++++++++++++++++++++++++
 arch/arm/dts/ast2700.dtsi     | 14 ++++++++++++++
 configs/evb-ast2700_defconfig |  7 ++++---
 drivers/misc/Makefile         |  2 +-
 4 files changed, 52 insertions(+), 4 deletions(-)

diff --git a/arch/arm/dts/ast2700-evb.dts b/arch/arm/dts/ast2700-evb.dts
index 648be447b82b..4edbc2a8e059 100644
--- a/arch/arm/dts/ast2700-evb.dts
+++ b/arch/arm/dts/ast2700-evb.dts
@@ -196,6 +196,39 @@
 #endif
 };
 
+&edaf_bridge0 {
+	status = "okay";
+	espi-base = <0x0 0x14c05000>;
+#if 0 //DDR mode using edaf0 reserved memory
+	edaf-ddr-mode;
+	mem-base = <&edaf0>;
+#else
+	mem-base = <0x1 0x80000000>; //SPI0
+#endif
+	ctl-base = <0x0 0x14010000>;
+	cmd-read = <0x130401>;
+	cmd-write-enable = <0x6041e>;
+	cmd-write = <0x120402>;
+	cmd-read-status = <0x50415>;
+	cmd-erase-4k = <0x21040e>;
+	cmd-erase-32k = <0x5c040e>;
+	cmd-erase-64k = <0xdc040e>;
+};
+
+&edaf_bridge1 {
+	status = "okay";
+	espi-base = <0x0 0x14c06000>;
+	mem-base = <0x2 0x00000000>; //SPI1
+	ctl-base = <0x0 0x14020000>;
+	cmd-read = <0x130401>;
+	cmd-write-enable = <0x6041e>;
+	cmd-write = <0x120402>;
+	cmd-read-status = <0x50415>;
+	cmd-erase-4k = <0x21040e>;
+	cmd-erase-32k = <0x5c040e>;
+	cmd-erase-64k = <0xdc040e>;
+};
+
 &emmc_controller {
 	status = "okay";
 	sdhci_hs200; /* enable hs200 */
diff --git a/arch/arm/dts/ast2700.dtsi b/arch/arm/dts/ast2700.dtsi
index f7987277bd8b..2ba713e10210 100644
--- a/arch/arm/dts/ast2700.dtsi
+++ b/arch/arm/dts/ast2700.dtsi
@@ -375,6 +375,20 @@
 			};
 		};
 
+		edaf_bridge0: edaf-bridge@14c38000 {
+			compatible = "aspeed,ast2700-edaf-bridge0";
+			reg = <0x0 0x14c38000 0x0 0x200>;
+			aspeed,scu1 = <&syscon1>;
+			status = "disabled";
+		};
+
+		edaf_bridge1: edaf-bridge@14c38200 {
+			compatible = "aspeed,ast2700-edaf-bridge1";
+			reg = <0x0 0x14c38200 0x0 0x200>;
+			aspeed,scu1 = <&syscon1>;
+			status = "disabled";
+		};
+
 		spi0: flash-controller@14010000 {
 			reg = <0x0 0x14010000 0x0 0xc4>,
 			      <0x1 0x80000000 0x0 0x80000000>;
diff --git a/configs/evb-ast2700_defconfig b/configs/evb-ast2700_defconfig
index 4d63c8450917..2be37a13191e 100644
--- a/configs/evb-ast2700_defconfig
+++ b/configs/evb-ast2700_defconfig
@@ -55,7 +55,6 @@ CONFIG_CMD_BOOTZ=y
 # CONFIG_CMD_IMI is not set
 # CONFIG_CMD_XIMG is not set
 CONFIG_CMD_EEPROM=y
-CONFIG_CMD_MISC=y
 CONFIG_CMD_MEMTEST=y
 CONFIG_SYS_ALT_MEMTEST=y
 # CONFIG_CMD_LZMADEC is not set
@@ -63,6 +62,7 @@ CONFIG_SYS_ALT_MEMTEST=y
 CONFIG_CMD_CLK=y
 CONFIG_CMD_GPIO=y
 CONFIG_CMD_I2C=y
+CONFIG_CMD_MISC=y
 CONFIG_CMD_MMC=y
 CONFIG_CMD_PCI=y
 CONFIG_CMD_SF_TEST=y
@@ -103,8 +103,10 @@ CONFIG_SYS_I2C_AST2600=y
 # CONFIG_INPUT is not set
 CONFIG_DM_MAILBOX=y
 CONFIG_AST2700_MBOX=y
-CONFIG_ASPEED_MBOX=y
 CONFIG_ASPEED_AHBC=y
+CONFIG_ASPEED_VGA=y
+CONFIG_ASPEED_EDAF_BRIDGE=y
+CONFIG_ASPEED_MBOX=y
 CONFIG_I2C_EEPROM=y
 CONFIG_SUPPORT_EMMC_BOOT=y
 CONFIG_MMC_IO_VOLTAGE=y
@@ -169,4 +171,3 @@ CONFIG_ECDSA_VERIFY=y
 CONFIG_TPM=y
 CONFIG_LZ4=y
 CONFIG_LZMA=y
-CONFIG_ASPEED_VGA=y
diff --git a/drivers/misc/Makefile b/drivers/misc/Makefile
index 266ca121fa95..a187155ce63f 100644
--- a/drivers/misc/Makefile
+++ b/drivers/misc/Makefile
@@ -92,7 +92,7 @@ obj-$(CONFIG_SL28CPLD) += sl28cpld.o
 obj-$(CONFIG_$(SPL_)ASPEED_DP) += aspeed_dp.o
 obj-$(CONFIG_$(SPL_)ASPEED_VGA) += aspeed_vga.o
 obj-$(CONFIG_$(SPL_)ASPEED_VBIOS) += aspeed_vbios.o
-obj-$(CONFIG_$(SPL_TPL_)ASPEED_EDAF_BRIDGE) += aspeed_edaf_bridge.o
+obj-$(CONFIG_ASPEED_EDAF_BRIDGE) += aspeed_edaf_bridge.o
 obj-$(CONFIG_SPL_SLI_AST2700) += sli_ast2700.o
 obj-$(CONFIG_SPL_PRIVILEGE_CTRL) += aspeed_prictrl.o aspeed_sram_prictrl.o
 obj-$(CONFIG_SPL_LTPI_AST2700) += ltpi_ast2700.o
-- 
2.34.1

